lib/matplotlib/axes/_axes.py

```diff
@@ -2749,13 +2749,13 @@ def get_next_color():
             if rotatelabels:
                 label_alignment_v = yt > 0 and 'bottom' or 'top'
                 label_rotation = np.rad2deg(thetam) + (0 if xt > 0 else 180)
-
-            t = self.text(xt, yt, label,
-                          size=rcParams['xtick.labelsize'],
-                          horizontalalignment=label_alignment_h,
+            props = dict(horizontalalignment=label_alignment_h,
                           verticalalignment=label_alignment_v,
                           rotation=label_rotation,
-                          **textprops)
+                          size=rcParams['xtick.labelsize'])
+            props.update(textprops)
```

Comment:
On Py<3.6 that's going to run into "interesting" issues when people pass shorthands such as `ha` in textprops (due to random dict iteration, either `ha` or `horizontalalignment` will win).  On Py3.6+ the latter will always win (guaranteed dict iteration order) and life's good.
OTOH the original implementation also had the same problems, so heh.

Reply:
Well I thought it shouldn't matter, since as I see it `ha` will always win. The reason is that (at least currently) `Text` takes `horizontalalignment` as named argument and `ha`, if provided, would be part of the **kwargs. Hence `horizontalalignment` will be set first within `Text.__init__` and later `ha` will overwrite it, due to the `self.update(kwargs)`. 

https://github.com/matplotlib/matplotlib/blob/3664c83425f86cbe4a2a5133514cee2366b2bfd1/lib/matplotlib/text.py#L185